pkg/server/api.go

```diff
@@ -134,10 +134,27 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
-	// we know we're at 3.2 in code.. serve directly, parsing is expensive...
-	// we're doing it during an HTTP request, and most notably before we write the HTTP headers
+
 	var serveConf *runtime.RawExtension
-	if reqConfigVer.Equal(*semver.New("3.2.0")) {
+	if reqConfigVer.Equal(*semver.New("3.3.0")) {
+		// If the config version is greater or equal to 3.3.0 we can add the kargs from the
+		// MachineConfig into the returned config
+		kargs, err := sh.server.GetKernelArguments(cr)
+		if err != nil {
+			w.Header().Set("Content-Length", "0")
+			w.WriteHeader(http.StatusInternalServerError)
+			glog.Errorf("couldn't get kernel arguments for req: %v, error: %v", cr, err)
```

Comment:
is there a reason we dont return here?

Reply:
Indeed it should. I'm not actually sure I like the code flow of fetching the config from https://github.com/openshift/machine-config-operator/pull/2675/files/513a02f4f53a5ac38bd5973f8d97da8ea510b58e#diff-9e2561386a5b88c1f09d35f98d4d4fa1cf4a3fb15a41183e21d5a42d2c24d939R125 and then re-feteching it in `GetKernelArguments`. I wanted to stay as true to Stephen's code and so I was approaching it from minizing changes. That said, since there is a chance for an error, it should be handled.

Reply:
Fixed up